Vslt/vslt 367 customer doc

xwa130saucer opened the pull request.

xwa130saucer committed:
linter readme

xwa130saucer committed:
update sauce connect doc

loyalchow commented on README.md:
````diff
@@ -40,12 +42,12 @@ module.exports = {
+npm run test-screener
 ```
 
 ### Testing Interactions
````
I see you've changed headings. This should also be `##` then

xwa130saucer replied:
addressed

loyalchow commented on README.md:
```diff
@@ -212,10 +219,6 @@ module.exports = {
-
-
-
 ### Additional Configuration Options
```
This should now also be `##`

xwa130saucer replied:
addressed

loyalchow commented on README.md:
````diff
@@ -287,47 +301,113 @@ module.exports = {
     }
     ```
+
+### Sauce Connect Integration
````
Please move this into Cross Browser Testing as a sub-section

xwa130saucer replied:
addressed

loyalchow commented on README.md:
```diff
@@ -287,47 +301,113 @@ module.exports = {
       extendedDebugging: true, // optional
-      tunnelIdentifier: 'MyTunnel01' // optional
+      tunnelIdentifier: 'MyTunnel01', // optional
+      launchSauceConnect: true // optional
```
Please add a note here to view "Sauce Connect" for more information

xwa130saucer replied:
addressed

loyalchow commented on README.md:
```diff
@@ -287,47 +301,113 @@ module.exports = {
+    accessKey: 'sauce_access_key',
+    maxConcurrent: 10, // optional available concurrency you have from Sauce Labs
+    extendedDebugging: true, // optional
+    launchSauceConnect: true // optional
```
Remove the `extendedDebugging` option to simplify this example

xwa130saucer replied:
addressed

loyalchow commented on README.md:
```diff
@@ -287,47 +301,113 @@ module.exports = {
+
+### Sauce Connect Integration
+
+You could run your tests through sauce connect tunnel by setting the flag `launchSauceConnect: true`:
```
"When using Sauce Labs browsers, you have the option to use the Sauce Connect tunnel by..."

"When enabled, Sauce Connect will be launched and managed by this module, and assigned a unique tunnel identifier."

xwa130saucer replied:
addressed

loyalchow commented on README.md:
````diff
@@ -287,47 +301,113 @@ module.exports = {
+  }
+  ```
+
+The sauce connect tunnel will generate `sauce-connect.log` log file for debugging purpose at the root directory of your `screener-runner` project.
````
Let's simplify this and the below, and simply add a "Sauce Connect Notes" section that includes a bullet on each:

* Must use Sauce browsers
* Where SC logs are stored
* That SC tunnel cannot be used the `tunnel` option
* SC ports
* Link to SC troubleshooting docs in Sauce Labs

xwa130saucer replied:
changed

xwa130saucer committed:
update doc

xwa130saucer committed:
update log file location

loyalchow commented on README.md:
````diff
@@ -212,69 +219,142 @@ module.exports = {
+  }
+  ```
 
+#### Sauce Connect Notes
````
Change to: `Important Notes on Sauce Connect`

xwa130saucer replied:
changed

loyalchow commented on README.md:
```diff
@@ -212,69 +219,142 @@ module.exports = {
 
+#### Sauce Connect Notes
 
+- Must use Sauce browsers
```
Please change this bullet to "Sauce Connect Integration requires all browsers to be Sauce Labs Browsers. An error is thrown when using non-Sauce browsers."

xwa130saucer replied:
changed

loyalchow commented on README.md:
```diff
@@ -212,69 +219,142 @@ module.exports = {
+- Must use Sauce browsers
 
-### Additional Configuration Options
+  When you run your tests in sauce labs, you could only run your tests inside sauce browsers, which means that you have to specify both the name of the browser and the version of the browser up to one digit after the decimal dot:
```
Please remove this and config

xwa130saucer replied:
changed

loyalchow commented on README.md:
```diff
@@ -212,69 +219,142 @@ module.exports = {
+
+  If you don't follow the rules, the runner will throw you the error: `Only Sauce Labs browsers with version can be used when launchSauceConnect flag is enabled`.
+
+- Where Sauce Connect logs are stored
```
Please change this to: "Logs for Sauce Connect are saved in the root of your project under `sauce-connect.log` for debugging purposes."

xwa130saucer replied:
changed

loyalchow commented on README.md:
```diff
@@ -212,69 +219,142 @@ module.exports = {
+
+  The sauce connect tunnel will generate `sauce-connect.log` log file for debugging purpose at the root directory of your project.
+
+- Sauce tunnel can not be used together with ngrok tunnel
```
Please change to: "Sauce Connect tunnel cannot be used together with the `tunnel` option."

xwa130saucer replied:
changed

loyalchow commented on README.md:
```diff
@@ -212,69 +219,142 @@ module.exports = {
+
+  when you enable `launchSauceConnect`. If you don't follow the rules, the runner will throw you the error: `tunnel option cannot be set when launchSauceConnect flag is enabled`.
+
+- `launchSauceConnect` and `tunnelIdentifier`
```
Please change to: "A unique `tunnelIdentifier` is automatically generated for you when using the Sauce Connect Integration. An error is thrown when `tunnelIdentifier` is set manually.

xwa130saucer replied:
changed

loyalchow commented on README.md:
```diff
@@ -212,69 +219,142 @@ module.exports = {
+
+  Please note that if you turn `launchSauceConnect` to be true, the runner will set a unique `tunnelIdentifier` for you insead of setting by yourself. If you don't follow the rules, the runner will throw you the error: `tunnelIdentifier cannot be set when launchSauceConnect flag is enabled`.
+
+- Sauce Connect valid ports
```
Please change to: "When running Sauce Connect tunnel on your localhost, please note that Sauce Connect only supports a limited set of valid ports."

xwa130saucer replied:
changed

loyalchow commented on README.md:
```diff
@@ -212,69 +219,142 @@ module.exports = {
+
+  When you run your tests in your localhost, please be cautious that sauce connect only supports these [valid ports](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+FAQS#SauceConnectProxyFAQs-CanIAccessApplicationsonlocalhost?).
+
+- Further question
```
Please change to: "For additional information on Sauce Connect please refer to the Sauce Connect FAQ and Sauce Connect Troubleshooting documentation."

xwa130saucer replied:
changed

xwa130saucer committed:
update doc

loyalchow commented on README.md:
```diff
@@ -212,69 +219,123 @@ module.exports = {
+- Sauce Connect Integration requires all browsers to be Sauce Labs Browsers. An error is thrown when using non-Sauce browsers.
 
-### Additional Configuration Options
+  If you don't follow the rules, the runner will throw you the error: `Only Sauce Labs browsers with version can be used when launchSauceConnect flag is enabled`.
```
Please remove this - I don't think it's necessary, as we mention an error above already

loyalchow commented on README.md:
```diff
@@ -212,69 +219,123 @@ module.exports = {
+
+- Logs for Sauce Connect are saved in the root of your project under `sauce-connect.log` for debugging purposes.
+
+  The sauce connect tunnel will generate `sauce-connect.log` log file for debugging purpose at the root directory of your project.
```
Please remove this. Line above covers it

loyalchow commented on README.md:
```diff
@@ -212,69 +219,123 @@ module.exports = {
+
+- Sauce Connect tunnel cannot be used together with the `tunnel` option.
+
+  `ngrok` tunnel is forbidden when you run the sauce connect tunnel, which means you can not set
```
Please remove this and below. Just keep first line of bullet. Clients do not know what "ngrok" is

loyalchow commented on README.md:
```diff
@@ -212,69 +219,123 @@ module.exports = {
+
+- A unique `tunnelIdentifier` is automatically generated for you when using the Sauce Connect Integration. An error is thrown when `tunnelIdentifier` is set manually.
+
+  Please note that if you turn `launchSauceConnect` to be true, the runner will set a unique `tunnelIdentifier` for you insead of setting by yourself. If you don't follow the rules, the runner will throw you the error: `tunnelIdentifier cannot be set when launchSauceConnect flag is enabled`.
```
Please remove this. Bullet line already covers

loyalchow commented on README.md:
```diff
@@ -212,69 +219,123 @@ module.exports = {
+
+- For additional information on Sauce Connect please refer to the Sauce Connect FAQ and Sauce Connect Troubleshooting documentation.
+
+  If you have further question, you could visit [Sauce Connect FAQ](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+FAQS) and [Sauce Connect troubleshooting](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+Troubleshooting) pages.
```
Please remove this, and add links to the first bullet line

loyalchow commented on README.md:
```diff
@@ -212,69 +219,123 @@ module.exports = {
+
+- When running Sauce Connect tunnel on your localhost, please note that Sauce Connect only supports a limited set of valid ports.
+
+  When you run your tests in your localhost, please be cautious that sauce connect only supports these [valid ports](https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+FAQS#SauceConnectProxyFAQs-CanIAccessApplicationsonlocalhost?).
```
Please remove this, and link the first bullet line "valid ports"

xwa130saucer committed:
update doc

xwa130saucer commented:
@loyalchow 
Updated all of the comments above. Sorry about letting you reword almost all of the write-ups...
I also updated the other repo (screener-storybook).
